constraints/miden-vm/memory.air

```diff
@@ -0,0 +1,78 @@
+mod MemoryAir
+
+### Helper functions/evaluators ###################################################################
+
+# Enforces that columns must be zero.
+ev equal_zero(&main: [4]):
+    enf v = 0 for v in &main
+
+# Enforces that values in the columns in the current row must be equal to the values in the next 
+# row.
+ev equal_seq_lines(&main: [4]):
+    enf v' = v for v in &main
+
+# Enforces that created flags and s[0] column have valid values during the program execution.
+ev flags_validity(main: [s_0], n0, n1, delta_a, delta_c):
+    # Enforce that n0 must be binary.
+    enf n0^2 = n0
+    # Enforce that when context changes, n0 = 1 (or when n0 = 0, context remains the same).
+    enf delta_c = 0 when !n0
+    # Enforce that n1 must be binary. An additional condition ensures that the check of n1 
+    # occurs only if the context does not change (n0 = 0).
+    enf n1^2 = n1 when !n0
+    # Enforce that if context remains the same, n1 = 1 when address changes and 0 otherwise.
+    enf delta_a = 0 when !n0 & !n1
```

Comment:
nit: Should we define `is_binary` evaluator instead of `enf n0^2 = n0` and enf `n1^2 = n1`?

nit: I think we should add spaces to improve readability, i.e.

```
    # Enforce that n0 must be binary.
    enf n0^2 = n0

    # Enforce that when context changes, n0 = 1 (or when n0 = 0, context remains the same).
    enf delta_c = 0 when !n0

    # Enforce that n1 must be binary. An additional condition ensures that the check of n1 
    # occurs only if the context does not change (n0 = 0).
    enf n1^2 = n1 when !n0

    # Enforce that if context remains the same, n1 = 1 when address changes and 0 otherwise.
    enf delta_a = 0 when !n0 & !n1
```

Reply:
As I understood. for now we decided to be strict and only pass columns to the evaluators, so I can't pass variables `n0` and `n1` to the evaluator.

Reply:
Ah, sorry, you're right. My bad.